Sources/Workspace/Workspace.swift

```diff
@@ -518,23 +507,46 @@ extension Workspace {
 
         // Resolve the dependencies.
         let updateResults = resolveDependencies(dependencies: updateConstraints, diagnostics: diagnostics)
-        guard !diagnostics.hasErrors else { return }
+        guard !diagnostics.hasErrors else { throw Diagnostics.fatalError }
 
         // Emit the time taken to perform dependency resolution.
         let resolutionDuration = Date().timeIntervalSince(resolutionStartTime)
         diagnostics.emit(data: WorkspaceDiagnostics.ResolverDurationNote(resolutionDuration))
 
-		// Update the checkouts based on new dependency resolution.
+        return (graphRoot: graphRoot, updateResults: updateResults)
+    }
+
+    /// Updates the current dependencies.
+    ///
+    /// - Parameters:
+    ///     - diagnostics: The diagnostics engine that reports errors, warnings
+    ///       and notes.
+    public func updateDependencies(root: PackageGraphRootInput, diagnostics: DiagnosticsEngine)  {
+        guard let (graphRoot, updateResults) = diagnostics.wrap({ try computePackageUpdateResult(root: root, diagnostics: diagnostics) }), !diagnostics.hasErrors else { return } 
+        // Update the checkouts based on new dependency resolution.
         updateCheckouts(root: graphRoot, updateResults: updateResults, updateBranches: true, diagnostics: diagnostics)
         guard !diagnostics.hasErrors else { return }
-
+        
+        // Abort if we're unable to load the pinsStore or have any diagnostics.
+        guard let pinsStore = diagnostics.wrap({ try pinsStore.load() }), !diagnostics.hasErrors else { return }
         // Update the pins store.
         return pinAll(
             graphRoot: graphRoot,
             pinsStore: pinsStore,
             diagnostics: diagnostics)
     }
 
+    /// List packages available for update.
+    ///
+    /// - Parameters:
+    ///     - diagnostics: The diagnostics engine that reports errors, warnings
+    ///       and notes.
+    public func updateDependenciesDryRun(root: PackageGraphRootInput, diagnostics: DiagnosticsEngine) -> [PackageReference: PackageStateChange] {
+        guard let (graphRoot, updateResults) = diagnostics.wrap({ try computePackageUpdateResult(root: root, diagnostics: diagnostics) }), !diagnostics.hasErrors  else { return [:] } 
+        guard let result = diagnostics.wrap({ try computePackageStateChanges(root: graphRoot, resolvedDependencies: updateResults , updateBranches: true) }), !diagnostics.hasErrors  else { return [:] } 
```

Comment:
It'll be better to wrap both of these method calls into one `diagnostics.wrap{...}`

Reply:
done